packages/core/upload/server/controllers/view-configuration.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const { getService } = require('../utils');
+const { ACTIONS, FILE_MODEL_UID } = require('../constants');
+const validateConfig = require('./validation/admin/configureView');
+
+module.exports = {
+  async updateViewConfiguration(ctx) {
+    const {
+      request: { body },
+      state: { userAbility },
+    } = ctx;
+
+    if (userAbility.cannot(ACTIONS.configureView, FILE_MODEL_UID)) {
+      return ctx.forbidden();
+    }
+
```

Comment:
I was looking at the content-manager as a reference and I am seeing each plugin deals with permissions a little bit different. For instance:

- content-manager has a permissions services, where it registers all the actions & has functions like canConfigureContentType. But in upload we have a constants file.

- Super minor thing, we say getViewConfiguration here and findViewConfiguration in there.


The part that I understand the less is how actions are registered in the upload plugin. And if there is a "standard"way of doing it in the repo.

Reply:
Thanks for the review

- Do you suggest we move to the permission checker service approach in the upload plugin? imo we use the `userAbility` in quite a simple way in the `upload` plugin so think calling `.cannot` directly is clear enough
- Good point I have renamed it to `findViewConfiguration `
- In terms of how the actions are registered you can see in `packages/core/admin/server/services/role.js` we have added the new permission `plugin::upload.configure-view`. L285. This is how I understand the functionality:
  - Every time strapi boots the function `resetSuperAdminPermissions` is run, granting super admins all permissions
  - `createRolesIfNoneExist` is responsible for assigning permissions to `editor` and `author` roles. This is only run on strapis first boot. Therefore the new CTV permission will not automatically be added to editors and authors but can be added through the Admin UI

Reply:
> Do you suggest we move to the permission checker service approach in the upload plugin? imo we use the userAbility in quite a simple way in the upload plugin so think calling .cannot directly is clear enough

Agree, I think it is super clean like you have now. So I would leave it as it is!

Thanks a lot for the actions explanation 🙇